### DISCUSSION TITLE
How to calculate the scroll distance using velocity?

### DISCUSSION THREAD

**Discussion by Deepak1799** [microsoft/WinUI-Gallery#1011] [Q&A]
I am using WinUI [Scroll Viewer](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.scrollviewer?view=winrt-22621) and want to disable its built-in manipulations to achieve my other requirements. So the default fling scrolling is not working.

What I am trying to achieve fling scrolling is, I am using the velocity obtained from [ManipulationInertiaStarting](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.scrollviewer?view=winrt-22621) event to detect fling action.

I am getting the proper velocity, but I am unable to find a proper calculation to find the distance to which the control should scroll using the velocity values.

The only thing I know is distance is directly proportional to velocity. So I am using an hard-coded proportion factor. Can anyone please advice me whether this is correct or there is any proper way (formula) to calculate the scroll distance using velocity?

**Code snippet:**
_Provide manipulate mode and wire inertia event_
```
scrollViewer.ManipulationMode = ManipulationModes.TranslateInertia | ManipulationModes.TranslateX | ManipulationModes.TranslateY;
scrollViewer.ManipulationInertiaStarting += OnManipulationInertiaStarting;
```
_Calculate distance On Event Handler_

```
       private void OnManipulationInertiaStarting(object sender, ManipulationInertiaStartingRoutedEventArgs e)
        {
            var CONSTANT = 8;

            var velocityX = e.Velocities.Linear.X;
            var velocityY = e.Velocities.Linear.Y;

            var horizontalDistanceToScroll = velocityX * dpi * CONSTANT;
            var verticalDistanceToScroll = velocityY * dpi * CONSTANT;
        }
```

Thank you in advance.

**Comment by marcelwgn** [comment]
I think the correct way (not tested though yet) would be to use the deceleration to calculate how long the inertia is taking effect to then calculate the distance based on that time. Maybe this 8 is the magic number for how long this takes but I have no idea where this is coming from. Do you happen to have a 120Hz display?

**Reply by Deepak1799** [reply]
Hi @chingucoding ,

Thanks for your reply. I am using 60Hz display. Can you please let me know on how to get the deceleration and whether display have any effect?

Regards,
Deepak G

**Reply by marcelwgn** [reply]
Hmm, I see. Let me see if there is any official documentation on the deceleration, otherwise you might need to reverse engineer this :/
